File: QuanLyCuaHangUXUI/QuanLyCuaHangProject/src/constants/formatDate.js
// Hàm định dạng ngày tháng
export default function formatDate(date) {
  if (!date) return ''
  const d = new Date(date)
  return d.toLocaleDateString('vi-VN', {
    year: 'numeric',
    month: '2-digit',
    day: '2-digit',
  })
}

File: APIQuanLyCuaHang/APIQuanLyCuaHang/Helpers/Utils/FormatDate.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using APIQuanLyCuaHang.Constants;

namespace APIQuanLyCuaHang.Helpers.Utils
{
    public static class FormatDate
    {
        public static Func<DateTime, bool> GetDateFilter(string timeRange, DateTime today)
        {
            switch (timeRange)
            {
                case "day":
                    return x => x.Date == today.Date;
                case "week":
                    return x => x.Month == today.Month && (x.Day - 1) / 7 + 1 == (today.Day - 1) / 7 + 1;
                case "month":
                    return x => x.Month == today.Month && x.Year == today.Year;
                case "year":
                    return x => x.Year == today.Year;
                default:
                    throw new ArgumentException("Invalid time range specified.");
            }
        }
    }
}